packer/builder/azure/arm/builder.go

```diff
@@ -57,28 +60,67 @@ func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packe
 		return nil, err
 	}
 
+	servicePrincipalTokenVault, err := b.createServicePrincipalTokenVault()
+	if err != nil {
+		return nil, err
+	}
+
 	ui.Message("Creating Azure Resource Manager (ARM) client ...")
-	azureClient, err := NewAzureClient(b.config.SubscriptionID, b.config.ResourceGroupName, b.config.StorageAccount, servicePrincipalToken)
+	azureClient, err := NewAzureClient(b.config.SubscriptionID, b.config.ResourceGroupName, b.config.StorageAccount, servicePrincipalToken, servicePrincipalTokenVault)
```

Comment:
can you give this a line break so that the full line is readable in the awesome GitHub UI?